### PR TITLE
Fix error in old PyTorch for KL and JS divergence

### PR DESCRIPTION
**Pull request by kzkadc**
Fixes https://github.com/pytorch/ignite/pull/3232#issuecomment-2063322746

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/metrics/kl_divergence.py [hunk 1]
```diff
@@ -91,8 +91,9 @@ def update(self, output: Sequence[torch.Tensor]) -> None:
 
     def _update(self, y_pred: torch.Tensor, y: torch.Tensor) -> None:
         y_pred = F.log_softmax(y_pred, dim=1)
-        y = F.log_softmax(y, dim=1)
-        kl_sum = F.kl_div(y_pred, y, log_target=True, reduction="sum")
+        y = F.softmax(y, dim=1)
+        # y_pred and y are expected to be log probabilities and probabilities, respectively.
+        kl_sum = F.kl_div(y_pred, y, reduction="sum")
```
[the remark this answers is not in the document]
Comment (kzkadc) [reply ignite/metrics/kl_divergence.py:96]: Thank you for your suggestion.
Seeing pytorch's implementation of kl_div, setting log_target=True seems to be stable.

https://github.com/pytorch/pytorch/blob/277ab8a4c0d483bc0166a7eaa482a57a5c42c381/aten/src/ATen/native/Loss.cpp#L238-L251

I think switching by version would be better as your suggestion.

File: ignite/metrics/js_divergence.py [hunk 2]
```diff
@@ -71,14 +72,24 @@ class JSDivergence(KLDivergence):
...
+        m_prob = (y_pred_prob + y_prob) / 2
         m_log = m_prob.log()
-        y_pred = F.log_softmax(y_pred, dim=1)
-        y = F.log_softmax(y, dim=1)
-        self._sum_of_kl += (
-            F.kl_div(m_log, y_pred, log_target=True, reduction="sum")
-            + F.kl_div(m_log, y, log_target=True, reduction="sum")
-        ).to(self._device)
+
+        if Version(torch.__version__) >= Version("1.6.0"):
```
Comment (vfdev-5) [inline ignite/metrics/js_divergence.py:80]: Do you think we can gain in perfs if we factorize this check in the global space:
```python

TORCH_VERSION_GT_160 = Version(torch.__version__) >= Version("1.6.0")

class JSDivergence(...):
    ...

    def _update(...):
        ...
        if TORCH_VERSION_GT_160:
             ...
```

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks @kzkadc !

**Comment by vfdev-5** [comment]
There are unrelated CI failures to fix later, so will be merging this PR.
